sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py

```diff
@@ -83,24 +83,57 @@ def obtain_token_by_refresh_token(self, scopes, refresh_token, **kwargs):
     def _build_pipeline(self, config=None, policies=None, transport=None, **kwargs):
         pass
 
-    def _process_response(self, response, scopes, now):
-        # type: (dict, Sequence[str], int) -> AccessToken
-        _raise_for_error(response)
-
-        # TokenCache.add mutates the response. In particular, it removes tokens.
-        response_copy = copy.deepcopy(response)
-
-        self._cache.add(event={"response": response, "scope": scopes, "client_id": self._client_id}, now=now)
-        if "expires_on" in response_copy:
-            expires_on = int(response_copy["expires_on"])
-        elif "expires_in" in response_copy:
-            expires_on = now + int(response_copy["expires_in"])
+    def _process_response(self, response, request_time):
+        # type: (PipelineResponse, int) -> AccessToken
+
+        content = ContentDecodePolicy.deserialize_from_http_generics(response.http_response)
+
```

Comment:
Why not put _raise_for_error(content) here?

Reply:
If the error is that we tried to use an invalid refresh token, this method must evict it from the cache.

Reply:
Remove invalid cache should be part of the error handling. Right? ( I mean to remove invalid cache should be in the _raise_for_error method?)

Reply:
All `_raise_for_error` does is format an exception message and raise. I wouldn't say it handles errors.